### ISSUE TITLE
prevent-abbreviations: event_ is suffixed with underscore

### ISSUE THREAD

**Issue by felixfbecker** [sindresorhus/eslint-plugin-unicorn#667]
Why is `event_` suffixed with underscore? 🤔 
```
Please rename the variable `e`. Suggested names are: `error`, `event_`. A more descriptive name will do too.
```

**Comment by fisker** [comment]
Can you check your code? I guess `event` is used somewhere.

**Comment by fisker** [comment]
This is expected, unless you didn't use `event`.

**Comment by felixfbecker** [comment]
I checked the code and there is no shadowed variable `event` anywhere in the scope. Here's the file/place where the rule complained:

https://github.com/sourcegraph/sourcegraph/blob/2b86b78820bf291c611bfea3493fcbab012d60bb/web/src/user/settings/accessTokens/UserSettingsCreateAccessTokenPage.tsx#L76

There is an `eventLogger` in scope, but not `event`

Could this be reopened?

**Comment by fisker** [comment]
There is a global `event`,  because you use https://github.com/sourcegraph/sourcegraph/blob/2b86b78820bf291c611bfea3493fcbab012d60bb/.eslintrc.js#L6

**Comment by fisker** [comment]
Actually, I thought about this before, do we really need check scope up?

**Comment by felixfbecker** [comment]
Ah, that makes sense. Hmm.

Could the suffixing only be done if the same-name variable is actually used in the scope? If it's not used, it shouldn't be necessary to give it a different name

**Comment by felixfbecker** [comment]
I.e. only suffix in a case like this:
```ts
let error
try {
  foo()
} catch (error_) {
  error = error_
}
```
but not in this case:
```ts
let error
try {
  foo()
} catch (error) {
  console.error(error)
}
```

**Comment by fisker** [comment]
Yes, it could done, but @bmish  bring the `no-shadow` rule up today https://github.com/sindresorhus/eslint-plugin-unicorn/pull/749#discussion_r430786045 .

I'm not sure what should we do

**Comment by felixfbecker** [comment]
I don't know how ESLint works, but could the rule detect whether `no-shadow` is used? Alternatively, have an option so the user can allow shadowing  specifically

**Comment by fisker** [comment]
I think we don't read use configs in any other rule. 

BTW: what's your opinion about child scopes?

```js
try {

} catch (x) {
//       ^ we have a rule fix this to error, currently we are fixing to `error`, not `error_`, should we check the child scope?
	try {
	
	} catch (error) {
	
	}
}

```

**Comment by felixfbecker** [comment]
Fine to fix to `error`, ***unless*** inside the second `catch` there is a reference to `x` (which there isn't).

**Comment by fisker** [comment]
Yep, I prefer `error` too, but again, against `no-shadow`(with option `all`).

**Comment by fisker** [comment]
Let's close this issue for now, we can discuss the [avoidCapture](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/rules/utils/avoid-capture.js) logic later.
